### ISSUE TITLE
The login page is not redirecting to the access controlled page?

### ISSUE THREAD

**Issue by bobrocke** [getgrav/grav-plugin-login#199]
I have a site with an access controlled page at `addpage-blog-post` which redirects to the login page, `authenticate`, as expected. But once logged in, the user is not redirected back to the `addpage-blog-post` as the default behavior suggests.

`authenticate.md` has `login_redirect_here: false` set. And the plug-in YAML configuration is default with `redirect_after_login:  `.

What have I misconfigured?

**Comment by otahirs** [comment]
**I am experiencing the same issue.**

half off-topic:
In short, is there a simple way to have a button/link navigating to a login page, and after successful login get redirected back to the page on which the login prompt was started?

(I am displaying additional content on a page when user is logged in, so no place basic for `access:
  site.login: true`)

**Comment by otahirs** [comment]
For anyone interested, I got it working by using custom login page "/login" and custom [php twig extension](https://learn.getgrav.org/cookbook/plugin-recipes), placed in _user/plugins/login/templates/login.html.twig_ 

login.html.twig

```
{% extends 'partials/base.html.twig' %}

{% block content %}
    {{phpLoginRedirect(grav.user.authenticated)}}
    {% include 'partials/messages.html.twig' %}
    {% include 'partials/login-form.html.twig' %}
{% endblock %}
```

the function
```
function LoginRedirect($authenticated){
        // placed in "user/plugins/login/templates/login.html.twig" - {{phpLoginRedirect(grav.user.authenticated)}}

        if(isset($_SERVER['HTTP_REFERER'])) {
            if(!$authenticated){
                $_SESSION['ref'] = $_SERVER['HTTP_REFERER'];
            }
            else{
                $login_path = "http://" . $_SERVER['HTTP_HOST'] . "/login";
                $refferer_path = $_SERVER['HTTP_REFERER'];

                if(isset($_SESSION['ref']) && $login_path == $refferer_path){
                    
                    header('Location: ' . $_SESSION['ref']);
                    exit();
                }
            }
        }  
    }
```

**Comment by bobrocke** [comment]
@otahirs Is this a universal fix? Is it worth offering as a pull request?

**Comment by otahirs** [comment]
@bobrocke It is **just a** dirty **workaround** as it is not using the plugin default files and rely on caling separate php code within a template and the _/login_ path is absolute to my setup. I made it after failing trying to understand the plugin own php code as I am not that experienced programmer. 

However, the idea behind it works and with a little bit of php knowledge could be easily implemented into the existing plugin.

**Comment by bobrocke** [comment]
I'm still having the problem after updating to Grav v1.6.0-rc.2 and Login v3.0.0-rc.2.

**Comment by lauroguedes** [comment]
Any news about resolving this bug? I have been having this problem since version 1.5. When logging in, it does not return to the page that was previously requested as it should be the default behavior of the plugin. Help us

**Comment by Romarain** [comment]
Same for me.

**Comment by rhukster** [comment]
Can you guys please test the commit above?

**Comment by lauroguedes** [comment]
Now, it worked perfectly, thank you.

**Comment by buzatuAda** [comment]
do you have an approximate date this fix will go in the stable release (3.0.3)?
